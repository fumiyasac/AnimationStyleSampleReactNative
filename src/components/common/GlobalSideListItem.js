/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  StyleSheet
} from 'react-native';

import {
  Text,
  ListItem
} from 'native-base';

export default class GlobalSideListItem extends React.Component {
  render() {
    const { title, onPressListItem } = this.props;
    return (
      <ListItem style={styles.listItem} onPress={onPressListItem}>
        <Text style={styles.listItemText}>{title}</Text>
      </ListItem>
    );
  };
}

const styles = StyleSheet.create({
  listItem: {
    height: 50
  },
  listItemText: {
    fontSize: 12,
    fontWeight: 'bold',
    color: '#555555'
  }
});
