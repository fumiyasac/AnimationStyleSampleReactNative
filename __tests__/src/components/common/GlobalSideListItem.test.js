import 'react-native';
import React from 'react';
import GlobalSideListItem from '../../../../src/components/common/GlobalSideListItem';

import renderer from 'react-test-renderer';

it('GlobalSideListItemコンポーネントが正しく表示されること', () => {

  let stubObject = { title: "フィード", selectedIndex: 0 };

  const onPressListItemMock = jest.fn();

  const tree = renderer.create(
    <GlobalSideListItem 
      key={0}
      title={stubObject.title}
      onPressListItem={onPressListItemMock()} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
  