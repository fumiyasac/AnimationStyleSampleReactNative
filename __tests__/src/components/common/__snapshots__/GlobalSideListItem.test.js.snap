// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalSideListItemコンポーネントが正しく表示されること 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    onPress={undefined}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 15,
          "paddingBottom": 13,
          "paddingRight": 15,
          "paddingTop": 13,
        },
        Object {
          "height": 50,
        },
      ]
    }
    virtual={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#555555",
            "fontSize": 12,
            "fontWeight": "bold",
          },
        ]
      }
      uppercase={false}
      virtual={undefined}
    >
      フィード
    </Text>
  </View>
</View>
`;
